### PR TITLE
Update block preview images, add READMEs and fix Algolia indexing

### PR DESCRIPTION
**Pull request by kachkaev**
Upstream PRs:
- https://github.com/hashintel/hash/pull/660 by @MaggieAppleton 
- https://github.com/hashintel/hash/pull/661 by @jtewright 

\+ two small bonuses (see comments)

### CODE REVIEW

**Comment by kachkaev** [comment]
I renamed to fix this warning in `yarn install`

```
Warning: patch-package detected a patch file version mismatch
  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:
  Patch file created for
    domhandler@4.2.2
  applied to
    domhandler@4.3.1
  
  At path
  
    node_modules/domhandler
  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run
    patch-package domhandler
  to update the version in the patch file name and make this warning go away.
```

**Comment by kachkaev** [comment]
I renamed `patches/domhandler+4.2.2.patch` to `patches/domhandler+4.3.1.patch` to fix this warning during `yarn install`

```
Warning: patch-package detected a patch file version mismatch
  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:
  Patch file created for
    domhandler@4.2.2
  applied to
    domhandler@4.3.1
  
  At path
  
    node_modules/domhandler
  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run
    patch-package domhandler
  to update the version in the patch file name and make this warning go away.
```

File: site/scripts/sync-algolia-index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import siteMap from "../site-map.json" assert { type: "json" };
 
 const monorepoRoot = path.resolve(
   path.dirname(fileURLToPath(import.meta.url)),
-  "..",
+  "../..",
```
Comment (kachkaev) [inline site/scripts/sync-algolia-index.ts:12]: We moved this file from `scripts` to `site/scripts` in https://github.com/blockprotocol/blockprotocol/pull/375, but I did not spot this new issue with `monorepoRoot`. cc @CiaranMn

File: hub/@hash/callout.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "repository": "https://github.com/hashintel/hash.git",
-  "commit": "c56a9ea4a16c0926b8b2b65390a95fab9b29ce6b",
+  "commit": "02f8050c38a276c2d7515fc23a64c605ba48277c",
```
Comment (kachkaev) [inline hub/@hash/callout.json:3]: https://github.com/hashintel/hash/commit/02f8050c38a276c2d7515fc23a64c605ba48277c

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**